hoss/dimension_utilities.py

```diff
@@ -75,8 +75,143 @@ def prefetch_dimension_variables(opendap_url: str, varinfo: VarInfoFromDmr,
+    dimension_data = dimension_dataset[dimension_path][:]
+
+    # Determine the dimension's resolution by taking the median value 
+    # of the differences between each ascending data value.
+    dimension_array = np.array(dimension_data)
+
+    # Build array.
+    size = dimension_array.size
+    min_max_pairs = [[dimension_array[idx], 
+                      dimension_array[idx+1]] 
+                      for idx in range(0, size-1)]
+
+    # Calculate final values.
+    dim_resolution = np.median(np.diff(dimension_array))
+    min_max_pairs.append([dimension_array[size-1], 
+                          dimension_array[size-1] + dim_resolution])
+
+    return np.array(min_max_pairs)
+
+
+def write_bounds(dimension_dataset: Dataset,
+                 dimension_variable: VariableFromDmr,
+                 min_and_max_bounds: np.ndarray) -> None:
+    """ Write the input bounds array to a given dimension dataset.
+        
+        First a new dimension is created for the new bounds variable
+        to allow the variable to be two-dimensional.
+        
+        Then the new bounds variable is created using two dimensions: 
+        (1) the existing dimension of the dimension dataset, and
+        (2) the new bounds variable dimension.
+    
+    """
+    # Create the second bounds dimension.
+    dimension_full_name_path = dimension_variable.full_name_path
+    dimension_group = '/' + '/'.join(dimension_full_name_path.split('/')[1:-1])
+    dimension_name = dimension_full_name_path.split('/')[-1]
+
+    # Consider the special case when the dimension group is the root directory. 
+    # The dimension can't refer to the full path in the name itself, so we have
+    # to create it with respect to the group we want to place it in.
+    if dimension_group == '/':
+        bounds_dim = dimension_dataset.createDimension(dimension_name + '_bnds_dim', 2)
+    else:
+        bounds_dim = dimension_dataset[dimension_group].createDimension(dimension_name + '_bnds_dim', 2)
+
+    # Dimension variables only have one dimension - themselves.
+    variable_dimension = dimension_dataset[dimension_full_name_path].dimensions[0]
+
+    bounds_data_type = str(dimension_variable.data_type)
+    bounds = dimension_dataset.createVariable(dimension_full_name_path + '_bnds',
+                                              bounds_data_type,
+                                              (variable_dimension,
+                                               bounds_dim,))
+    # Write data to dataset file.
+    size = len(min_and_max_bounds)
+    for idx in range(0, size):
+        bounds[idx, 0] = (min_and_max_bounds[idx])[0]
+        bounds[idx, 1] = (min_and_max_bounds[idx])[1]
+
+    # Update varinfo attributes and references.
+    bounds_name = dimension_name + '_bnds'
+    dimension_dataset[dimension_full_name_path].setncatts({'bounds': bounds_name})
+    dimension_variable.references['bounds'] = {bounds_name,}
+    dimension_variable.attributes['bounds'] = bounds_name
```

Comment:
I know we (me and @owenlittlejohns talked about adding a unit attribute/reference here, but I was thinking it isn't really necessary since everything is happening under the hood?

Reply:
Thinking about it some more, we're probably okay without a `units` attribute on the `bounds` variable. I don't think it's being checked at all for any functional reason, and it's an ephemeral file, so the end user will never see it.